src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/DependencyTreeFlags.cs

```diff
@@ -80,53 +55,61 @@ public static class DependencyTreeFlags
         public static readonly ProjectTreeFlags UnresolvedDependencyFlags = Unresolved + DependencyFlags;
         public static readonly ProjectTreeFlags ResolvedDependencyFlags = Resolved + DependencyFlags;
 
+        /// <summary>
+        /// The set of flags to assign to unresolved Reference nodes.
+        /// </summary>
+        /// <remarks>
+        /// Contains <see cref="ProjectTreeFlags.Common.BrokenReference"/> which stops
+        /// <c>IGraphProvider</c> APIs from being called for that node.
+        /// </remarks>
         internal static readonly ProjectTreeFlags GenericUnresolvedDependencyFlags
-                = UnresolvedDependencyFlags
-                + UnresolvedReferenceFlags
+                = ProjectTreeFlags.Reference
+                + ProjectTreeFlags.BrokenReference 
+                + UnresolvedDependencyFlags
                 + GenericDependency;
 
+        /// <summary>
+        /// The set of flags to assign to resolved Reference nodes.
+        /// </summary>
         internal static readonly ProjectTreeFlags GenericResolvedDependencyFlags
-                = ResolvedDependencyFlags
-                + ResolvedReferenceFlags
+                = ProjectTreeFlags.Reference
+                + ProjectTreeFlags.ResolvedReference
+                + ResolvedDependencyFlags
                 + GenericDependency;
 
-        internal static readonly ProjectTreeFlags TargetNode = ProjectTreeFlags.Create("TargetNode");
+        internal static readonly ProjectTreeFlags DependencyTargetFrameworkGroup = ProjectTreeFlags.Create("DependencyTargetFrameworkGroup");
```

Comment:
This name would change if we chose a different grouping here, I would avoid baking in "TargetFramework".

Reply:
Without an alternative I'm not sure what you mean. We currently have specific names for the other kinds of groupings we have here. I agree we may wish to restructure the tree at some point, however if we do that we can also modify the flags we use.

Reply:
#5961